### PR TITLE
Fix README.md files to be like man pages

### LINKED ISSUE #673: Manpage Contents should follow Conventions
Manpages have a fairly consistent formatting pattern that we should follow in our own.

This page has a pretty good example: http://www.tldp.org/HOWTO/Man-Page/q3.html

Specifically I propose we make the following changes:

1. Include a brief summary of the available options in the SYNOPSIS section. 
(ex. `git [--version] [--help] [-C <path>] [-c <name>=<value>] ...` )

2. Shorten our DESCRIPTION section. In the initial intent to create command descriptions this was appropriate. We could perhaps include some of the information in other sections.

3. Create an OPTIONS section for each of the commands. The purpose of this section would be to describe each of the available options in detail. I expect this is where some of the information in DESCRIPTION could land.

4. Add and ENVIRONMENT section to describe the environment variables which can interact with each of the commands. On the driver manpage, we may consider including manpages which interact with CoreHost.

cc @piotrpMSFT @blackdwarf

### PR DESCRIPTION
**Pull request by blackdwarf**
Change the SYNOPSIS section. Add ENVIRONMENT section where needed.
Change the driver's man page to include all of the current commands.
The MD files now conform to the man pages outline. The manpages are updated as
well.

Fix #673 

/cc @brthor

### CODE REVIEW

File: Documentation/manpages/dotnet-compile.1 [hunk 1]
```diff
@@ -64,65 +58,109 @@ The static library [staticlib] option has the default extension of ".a"
...
+Show\ short\ help.\ 
+\f[]
+.fi
+.SH ENVIRONMENT
+.PP
+\f[C]DOTNET_HOME\f[]
+.IP
+.nf
+\f[C]
+Points\ to\ the\ runtime\ that\ will\ be\ used\ to\ run\ the\ executable\ file\ that\ is\ dropped\ after\ compiling.\ Not\ needed\ for\ native\ compilation.\ \
```
Comment (brthor) [inline Documentation/manpages/dotnet-compile.1]: DOTNET_HOME points to the base directory with `bin` and `runtimes` in it.

File: Documentation/manpages/dotnet-compile.1 [hunk 2]
```diff
@@ -64,65 +58,109 @@ The static library [staticlib] option has the default extension of ".a"
...
+.nf
+\f[C]
+Points\ to\ the\ runtime\ that\ will\ be\ used\ to\ run\ the\ executable\ file\ that\ is\ dropped\ after\ compiling.\ Not\ needed\ for\ native\ compilation.\ \ 
+\f[]
+.fi
+.SH SEE ALSO
+.PP
+dotnet\-restore(1), dotnet\-publish(1), dotnet(1)
+.SH AUTHORS
+Zlatko Knezevic.
```
Comment (brthor) [inline Documentation/manpages/dotnet-compile.1]: Should you include an email here?

File: src/Microsoft.DotNet.Tools.Compiler.Csc/README.md [hunk 3]
```diff
@@ -24,28 +24,47 @@ The IL exe output type needs a runtime host to execute. The IL exe output type a
...
 
-This command relies on the following artifacts: source files, project.json project file, project.lock.json temporary file and restored NuGet dependencies. 
+This command relies on the following artifacts: source files, project.json project file and the "lock" file (project.lock.json). Prior to invoking dotnet-compile, dotnet-restore(1) should be run to restore any dependencies that are needed for the application.  
+
+# OPTIONS 
 
-The project.json file represents and describes the project. It can contain several setting, which are described in the [Build](https://docs.asp.net/en/latest/dnx/projects.html#building). The most important information in the project.json file are the root (not transitive) NuGet dependencies and the files to be compiled. By default, this is a wildcard -- "*.cs". It supports both inclusion and exclusion semantics.
+`-n, --native`
```
Comment (brthor) [inline src/Microsoft.DotNet.Tools.Compiler.Csc/README.md]: Is native an option for `dotnet-compile-csc` ?

**Comment by brthor** [comment]
Nice, the format seems much more manpage like :+1: 

Only thing I would like to see is an options section for `dotnet-compile`. It is probably going to be a popular command.

File: Documentation/manpages/dotnet-compile.1 (hunk 1 above)
Comment (blackdwarf) [inline Documentation/manpages/dotnet-compile.1]: Good point.

File: Documentation/manpages/dotnet-compile.1 (hunk 2 above)
Comment (blackdwarf) [inline Documentation/manpages/dotnet-compile.1]: Meh. I guess I should, it is usually done that way.

**Comment by blackdwarf** [comment]
@brthor I thought the compiler README.md has an "OPTIONS" section?

**Comment by brthor** [comment]
:smile: Must have missed it yesterday, sorry about that. Not sure what I was looking at

**Pull request merged**